### ISSUE TITLE
Add original TX Hash to message

### ISSUE THREAD

**Issue by tboeckmann** [mailchain/mailchain#293]
**Is your feature request related to a problem? Please describe.**
I want to have an option in the UI to be able to see the "origin" txhash of the message.

**Describe the solution you'd like**

Something akin to a light version of email client "show original", showing the fields from rpc getTransactionByHash (or equiv) [example tx](https://etherscan.io/tx/0x2bf261a81e624d649450a3851df2d0639a8b98ed3bce39da14e8f318adf3edb9).

The following would be added to the `message` `headers` object returned via the API. This aligns with the optional headers guidelines for RFC2822 header standards https://tools.ietf.org/html/rfc2822#section-3.6.8 ).

```
"transaction": {
  "blockHash": "0x94fa39de3950f87d9225d2146b19981e5824f271d1049348462cc2ee2f20f94f",
  "blockNumber": "0x81a753",
  "from": "0x4ad2b251246aafc2f3bdf3b690de3bf906622c51",
  "gas": "0x77c4",
  "gasPrice": "0x4a817c800",
  "hash": "0x2bf261a81e624d649450a3851df2d0639a8b98ed3bce39da14e8f318adf3edb9",
  "input": "0x6d61696c636861696e010a82012ee10c59024c836d7ca12470b5ac74673002127ddedadbc6fc4375a8c086b650060ede199f603a158bc7884a903eadf97a2dd0fbe69ac81c216830f94e56b847d924b51a7d8227c80714219e6821a51bc7cba922f291a47bdffe29e7c3f67ad908ff377bfcc0b603007ead4bfd87ff0acc272528ca03d6381e6d0e1e2c5dfd24d521",
  "nonce": "0x8",
  "to": "0x92d8f10248c6a3953cc3692a894655ad05d61efb",
  "transactionIndex": "0x40",
  "value": "0x0"
}
```
NOTE: The fields above map intersection fields from [Geth](https://github.com/ethereum/wiki/blob/78990aca283f67ceb26ed70c8eb82008424aaf64/JSON-RPC.md#eth_gettransactionbyhash) & [Parity](https://wiki.parity.io/JSONRPC-eth-module#eth_gettransactionbyhash) getTransactionByHash calls.

**Describe alternatives you've considered**

Considered including the encrypted message object. This would be nice to have.

**Additional context**

**Comment by robdefeo** [comment]
Include `block-number`,  `transaction-hash`, `block-number-encoding`  and `transaction-hash-encoding` on the response for get messages. 

This involves returning a struct instead of just the encrypted content in ```
// Receiver gets encrypted data from blockchain.
type Receiver interface {
	Receive(ctx context.Context, network string, address []byte) ([]cipher.EncryptedContent, error)
}```
The struct would be something like ```
type Transaction struct {
   Data []cipher.EncryptedContent
   BlockNumber []byte
   Hash []byte
}
```

**Comment by nikosk686** [comment]
Hi @robdefeo. This seems pretty straightforward, I believe I can have it done during next week.

What is not clear to me, though, is the meaning of `block-number-encoding` and `transaction-hash-encoding`.

Also I see that you propose []byte for the new fields, however the related `txResult` struct stores those field values as string. Do you mean that the []byte will contain both the original value and the encoding one?

**Comment by robdefeo** [comment]
Hi @nikosk686 That's excellent 😄 

We are looking to include `encoding-type` into all of the http responses, as bytes need to be encoded by something usually `hex` but protocols such as substrate use a mixture of `hex` and `base58` so we need to be explicit.  However the choice of encoding would happen when retuning the response in the API layer. The encoding consts are not yet merged to master (I hope to complete this in the next few days) https://github.com/mailchain/mailchain/blob/39991b28fbeab0b499314e246eef209b9dcbcde6/internal/encoding/consts.go#L31-L36

For the interface `[]byte` is needed as it allows any value, the `[]byte` contains only the value not the encoding.

**Comment by robdefeo** [comment]
@nikosk686 this changes have now been merge so you have access to them in the main branch. Hopefully this should unblock you.  This merge does mean that there are changes required in the UI to make every work and you will need to re-add your private keys if you are building from code. The work was largely done to the keystore to support different key types

**Comment by nikosk686** [comment]
Hi @robdefeo thanks for the update.
I will pull the changes and continue working on this.

**Comment by robdefeo** [comment]
@nikosk686 😄looking forward your pull request. In the meantime if there is anything I can do to help, just reach out.

**Comment by nikosk686** [comment]
Hi @robdefeo, I have pushed some changes related to this.
I am not sure I have understand the context completely, so 
I' ve created a pr, though it is wip, for you to have a first look
so we can catch early any misunderstandings from my side.

Regarding the use of `[]byte` in the interface, while I believe understand
the concept, in the specific case seems unreasonable as we would have to
cast from string at first and then to cast again back to string for the api
response, without any other modification taking place.
Probably if the Transaction struct gets moved somewhere more centrally
we can change it.

Also while I made changes in various files no tests failed, meaning that the
specific flow is untested. I will add tests needed for the final version of this pr.

**Comment by robdefeo** [comment]
Hi @nikosk686 glad you created the PR for this and we can work through it more iteratively. Your reasoning is correct in this case, however we are integrating new protocols and using a `[]byte` removes this problem.  The other reason for requesting a `[]byte` is that we will allow the front end to in some cases request the encoding type to use, making `[]byte` the raw data that can be encoded as needed. 

In the PR I will leave some feedback on the `Transaction` struct.

You are correct about the missing test for this part of the code base. This is something that certainly needs to be resolved. I will raise a separate issue if you prefer to keep this PR more tightly scoped.

**Comment by nikosk686** [comment]
Hi @robdefeo thanks for the comments, i ll have a look asap.
unless you need this merge to be done quickly I prefer to have tests done
in this pr so not to leave tech debt behind.

**Comment by robdefeo** [comment]
@nikosk686 very happy to wait and have you put the tests inplace. Putting tests in place will be a big benefit and make the PR more reliable process. Thank you!! 👍 👍

**Comment by robdefeo** [comment]
closed by #340
